src/main/java/org/openstreetmap/atlas/checks/event/CheckFlagEvent.java

```diff
@@ -105,17 +216,21 @@ public static JsonObject flagToFeature(final CheckFlag flag,
      * @return {@link JsonObject} created from {@link CheckFlag}
      */
     public static JsonObject flagToJson(final CheckFlag flag,
-            final Map<String, String> additionalProperties)
+            final Map<String, Object> additionalProperties)
     {
-        final JsonObject flagJson = GEOJSON_BUILDER.create(flag.getLocationIterableProperties())
+        final JsonObject flagJson = GEOJSON_BUILDER
+                .createFromGeometriesWithProperties(flag.getLocationIterableProperties())
                 .jsonObject();
         final JsonObject flagPropertiesJson = new JsonObject();
         flagPropertiesJson.addProperty("id", flag.getIdentifier());
         flagPropertiesJson.addProperty("instructions", flag.getInstructions());
-
         // Add additional properties
-        additionalProperties.forEach(flagPropertiesJson::addProperty);
-
+        additionalProperties
+                .forEach((key, value) -> flagPropertiesJson.addProperty(key, (String) value));
+        // Add relation properties of the flaggedRelation to all its members.
+        // If relations are not flagged then the list is empty and no action will be taken.
+        flag.getFlaggedRelations().stream().map(flaggedRelation -> flaggedRelation.getProperties())
```

Comment:
I'm not sure about adding relation properties to the feature collection properties. These should contain information that pertain to the whole flag. It would be better if each relation could either be a feature or if the properties were sub properties in the feature collection properties (like featurePropetries in the geojson).

Reply:
Is there a case where there would be more than one relation nested inside the `"relation"` property? 
The relation id should not be baked into the key, ideally, you would just add that single relations' properties inside the `"relation"` prop. 

relation: {
  wheelchair: "yes",
  ref: "259b",
  id: 687780
}

And `id` would be one of the fields

Reply:
If you flag more than 1 relation yes, as would be the case in the proposed AOIPolygon enhancements, or the current version of ShadowDetection. However, because there is the `"relation"` parent key, the sub key could just be the id.

Reply:
A flag is created  for a single object or a set of AtlasObjects using the createFlag method in BaseCheck. ``` CheckFlag createFlag(final Set<AtlasObject> objects, final String instruction)```.
Hence if multiple relations are used to create a flag, then we need the properties of each relations separately. I have not found a check that uses a set of AtlasObjects to create flags yet but it is possible to have multiple relations.

Reply:
Like @Bentleysb mentioned, I could simply give the relation id as the sub key instead of appending the id to "relation".

Reply:
If thats the case, then `"relations"` should be an array; and each object of that array can be represent a single Relation

Reply:
Just to be clear, in the above case, if there are multiple relations, then it would be show up as
<img width="451" alt="screen shot 2018-11-05 at 2 41 05 pm" src="https://user-images.githubusercontent.com/42149840/48031380-e0a99c00-e108-11e8-83e6-bc4c59f5ef28.png">

Reply:
This has been updated. The key "relations" now has JsonArray as its value with each element being the tags of each relations.